// @vkontakte/vkui/dist/components/PopoutWrapper/PopoutWrapper.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _getClassName = _interopRequireDefault(require("../../helpers/getClassName"));

var _classNames = _interopRequireDefault(require("../../lib/classNames"));

var _platform = require("../../lib/platform");

var _transitionEvents = _interopRequireDefault(require("../../lib/transitionEvents"));

var _withPlatform = _interopRequireDefault(require("../../hoc/withPlatform"));

var _dom = require("../../lib/dom");

var PopoutWrapper = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(PopoutWrapper, _Component);

  function PopoutWrapper(props) {
    var _this;

    (0, _classCallCheck2.default)(this, PopoutWrapper);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(PopoutWrapper).call(this, props));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "elRef", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "animationFinishTimeout", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onFadeInEnd", function (e) {
      if (!e || e.animationName === 'animation-full-fade-in') {
        _this.setState({
          opened: true
        });
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "preventTouch", function (e) {
      return e.preventDefault();
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onClick", function (e) {
      if (_this.state.opened && _this.props.onClick) {
        _this.props.onClick(e);
      }
    });
    _this.state = {
      opened: false
    };
    _this.elRef = _react.default.createRef();
    return _this;
  }

  (0, _createClass2.default)(PopoutWrapper, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (_dom.canUseDOM) {
        window.addEventListener('touchmove', this.preventTouch, {
          passive: false
        });
        this.waitAnimationFinish(this.elRef.current, this.onFadeInEnd);
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      // Здесь нужен последний аргумент с такими же параметрами, потому что
      // некоторые браузеры на странных вендорах типа Meizu не удаляют обработчик.
      // https://github.com/VKCOM/VKUI/issues/444
      if (_dom.canUseDOM) {
        // @ts-ignore (В интерфейсе EventListenerOptions нет поля passive)
        window.removeEventListener('touchmove', this.preventTouch, {
          passive: false
        });
        window.clearTimeout(this.animationFinishTimeout);
      }
    }
  }, {
    key: "waitAnimationFinish",
    value: function waitAnimationFinish(elem, eventHandler) {
      if (_transitionEvents.default.supported) {
        var eventName = _transitionEvents.default.prefix ? _transitionEvents.default.prefix + 'AnimationEnd' : 'animationend';
        elem.removeEventListener(eventName, eventHandler);
        elem.addEventListener(eventName, eventHandler);
      } else {
        var platform = this.props.platform;
        this.animationFinishTimeout = window.setTimeout(eventHandler, platform === _platform.ANDROID ? 300 : 600);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          alignY = _this$props.alignY,
          alignX = _this$props.alignX,
          closing = _this$props.closing,
          children = _this$props.children,
          hasMask = _this$props.hasMask,
          onClick = _this$props.onClick,
          className = _this$props.className,
          platform = _this$props.platform,
          restProps = (0, _objectWithoutProperties2.default)(_this$props, ["alignY", "alignX", "closing", "children", "hasMask", "onClick", "className", "platform"]);
      var baseClassNames = (0, _getClassName.default)('PopoutWrapper', platform);
      return _react.default.createElement("div", (0, _extends2.default)({}, restProps, {
        className: (0, _classNames.default)(baseClassNames, "PopoutWrapper--v-".concat(alignY), "PopoutWrapper--h-".concat(alignX), {
          'PopoutWrapper--closing': closing
        }, className),
        onClick: this.onClick,
        ref: this.elRef
      }), hasMask && _react.default.createElement("div", {
        className: "PopoutWrapper__mask"
      }), _react.default.createElement("div", {
        className: "PopoutWrapper__container"
      }, children));
    }
  }]);
  return PopoutWrapper;
}(_react.Component);

(0, _defineProperty2.default)(PopoutWrapper, "defaultProps", {
  hasMask: true,
  alignY: 'center',
  alignX: 'center',
  closing: false
});

var _default = (0, _withPlatform.default)(PopoutWrapper);

exports.default = _default;
