// @vkontakte/vkui/dist/es6/components/ConfigProvider/ConfigProvider.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React from 'react';
import PropTypes from 'prop-types';
import vkBridge from '@vkontakte/vk-bridge';
import { canUseDOM } from '../../lib/dom';
import ConfigProviderContext from './ConfigProviderContext';

var ConfigProvider = /*#__PURE__*/function (_React$Component) {
  _inherits(ConfigProvider, _React$Component);

  function ConfigProvider(props, context) {
    var _this;

    _classCallCheck(this, ConfigProvider);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(ConfigProvider).call(this, props));

    if (canUseDOM) {
      _this.setScheme((context.document || window.document).body, _this.mapOldScheme(props.scheme));
    }

    return _this;
  }

  _createClass(ConfigProvider, [{
    key: "mapOldScheme",
    value: function mapOldScheme(scheme) {
      switch (scheme) {
        case 'client_light':
          return 'bright_light';

        case 'client_dark':
          return 'space_gray';

        default:
          return scheme;
      }
    }
  }, {
    key: "setScheme",
    value: function setScheme(target, scheme) {
      target.setAttribute('scheme', scheme);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (prevProps.scheme !== this.props.scheme) {
        this.setScheme(this.document.body, this.mapOldScheme(this.props.scheme));
      }
    }
  }, {
    key: "getChildContext",
    value: function getChildContext() {
      return {
        isWebView: this.props.isWebView,
        webviewType: this.props.webviewType,
        scheme: this.mapOldScheme(this.props.scheme),
        app: this.props.app
      };
    }
  }, {
    key: "render",
    value: function render() {
      return React.createElement(ConfigProviderContext.Provider, {
        value: this.getChildContext()
      }, this.props.children);
    }
  }, {
    key: "document",
    get: function get() {
      return this.context.document || window.document;
    }
  }]);

  return ConfigProvider;
}(React.Component);

_defineProperty(ConfigProvider, "childContextTypes", {
  isWebView: PropTypes.bool,
  scheme: PropTypes.string,
  webviewType: PropTypes.oneOf(['vkapps', 'internal']),
  app: PropTypes.string
});

_defineProperty(ConfigProvider, "defaultProps", {
  webviewType: 'internal',
  isWebView: vkBridge.isWebView(),
  scheme: 'bright_light'
});

_defineProperty(ConfigProvider, "contextTypes", {
  document: PropTypes.object
});

export { ConfigProvider as default };
