// @vkontakte/vkui/dist/es6/components/Search/Search.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React, { Component } from 'react';
import classNames from '../../lib/classNames';
import withPlatform from '../../hoc/withPlatform';
import getClassname from '../../helpers/getClassName';
import Icon16SearchOutline from '@vkontakte/icons/dist/16/search_outline';
import Icon16Clear from '@vkontakte/icons/dist/16/clear';
import { IOS } from '../../lib/platform';
import Touch from '../Touch/Touch';
var searchId = 0;

var Search = /*#__PURE__*/function (_Component) {
  _inherits(Search, _Component);

  function Search(props) {
    var _this;

    _classCallCheck(this, Search);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Search).call(this, props));

    _defineProperty(_assertThisInitialized(_this), "isControlledOutside", void 0);

    _defineProperty(_assertThisInitialized(_this), "inputEl", void 0);

    _defineProperty(_assertThisInitialized(_this), "searchId", void 0);

    _defineProperty(_assertThisInitialized(_this), "onFocus", function (e) {
      _this.setState({
        focused: true
      });

      _this.props.onFocus && _this.props.onFocus(e);
    });

    _defineProperty(_assertThisInitialized(_this), "onBlur", function (e) {
      _this.setState({
        focused: false
      });

      _this.props.onBlur && _this.props.onBlur(e);
    });

    _defineProperty(_assertThisInitialized(_this), "onChange", function (e) {
      var target = e.target;

      if (!_this.isControlledOutside) {
        _this.setState({
          value: target.value
        });
      }

      if (_this.props.onChange) {
        _this.props.onChange(e);
      }
    });

    _defineProperty(_assertThisInitialized(_this), "onCancel", function () {
      // eslint-disable-next-line @typescript-eslint/unbound-method
      var nativeInputValueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, 'value').set;
      nativeInputValueSetter.call(_this.inputEl, '');
      var ev2 = new Event('input', {
        bubbles: true
      });

      _this.inputEl.dispatchEvent(ev2);
    });

    _defineProperty(_assertThisInitialized(_this), "onIconClickStart", function (e) {
      _this.props.onIconClick && _this.props.onIconClick(e.originalEvent);
    });

    _defineProperty(_assertThisInitialized(_this), "onIconCancelClickStart", function (e) {
      e.originalEvent.preventDefault();

      _this.inputEl.focus();

      _this.onCancel();
    });

    _defineProperty(_assertThisInitialized(_this), "inputRef", function (element) {
      var getRef = _this.props.getRef;
      _this.inputEl = element;

      if (getRef) {
        if (typeof getRef === 'function') {
          getRef(element);
        } else {
          getRef.current = element;
        }
      }
    });

    var state = {
      focused: false
    };

    if (props.hasOwnProperty('value')) {
      _this.isControlledOutside = true;
    } else {
      state.value = props.defaultValue || '';
    }

    _this.searchId = "search-".concat(searchId++);
    _this.state = state;
    return _this;
  }

  _createClass(Search, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          onFocus = _this$props.onFocus,
          onBlur = _this$props.onBlur,
          onChange = _this$props.onChange,
          defaultValue = _this$props.defaultValue,
          value = _this$props.value,
          placeholder = _this$props.placeholder,
          after = _this$props.after,
          getRef = _this$props.getRef,
          platform = _this$props.platform,
          icon = _this$props.icon,
          onIconClick = _this$props.onIconClick,
          inputProps = _objectWithoutProperties(_this$props, ["className", "onFocus", "onBlur", "onChange", "defaultValue", "value", "placeholder", "after", "getRef", "platform", "icon", "onIconClick"]);

      return React.createElement("div", {
        className: classNames(getClassname('Search', platform), {
          'Search--focused': this.state.focused,
          'Search--has-value': !!this.value,
          'Search--has-after': !!after,
          'Search--has-icon': !!icon
        }, className)
      }, React.createElement("div", {
        className: "Search__in"
      }, React.createElement("div", {
        className: "Search__width"
      }), React.createElement("div", {
        className: "Search__control"
      }, React.createElement("input", _extends({}, inputProps, {
        ref: this.inputRef,
        type: "text",
        className: "Search__input",
        onFocus: this.onFocus,
        onBlur: this.onBlur,
        onChange: this.onChange,
        value: this.value
      })), platform === IOS && after && React.createElement("div", {
        className: "Search__after-width"
      }, after), React.createElement("div", {
        className: "Search__placeholder"
      }, React.createElement("div", {
        className: "Search__placeholder-in"
      }, React.createElement(Icon16SearchOutline, null), React.createElement("div", {
        className: "Search__placeholder-text",
        dangerouslySetInnerHTML: {
          __html: placeholder
        }
      })))), React.createElement("div", {
        className: "Search__after",
        onClick: this.onCancel
      }, React.createElement("div", {
        className: "Search__icons"
      }, icon && React.createElement(Touch, {
        onStart: this.onIconClickStart,
        className: "Search__icon"
      }, icon), !!this.value && React.createElement(Touch, {
        onStart: this.onIconCancelClickStart,
        className: "Search__icon"
      }, React.createElement(Icon16Clear, null))), platform === IOS && after && React.createElement("div", {
        className: "Search__after-in"
      }, after))));
    }
  }, {
    key: "value",
    get: function get() {
      return this.isControlledOutside ? this.props.value : this.state.value;
    }
  }]);

  return Search;
}(Component);

_defineProperty(Search, "defaultProps", {
  autoComplete: 'off',
  placeholder: 'Поиск',
  after: 'Отмена'
});

export default withPlatform(Search);
