// @vkontakte/vkui/dist/es6/components/ModalPage/ModalPage.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React, { Component } from 'react';
import getClassName from '../../helpers/getClassName';
import classNames from '../../lib/classNames';
import withInsets from '../../hoc/withInsets';
import { isNumeric } from '../../lib/utils';
import withModalRootContext from '../ModalRoot/withModalRootContext';
import withPlatform from '../../hoc/withPlatform';

var ModalPage = /*#__PURE__*/function (_Component) {
  _inherits(ModalPage, _Component);

  function ModalPage() {
    _classCallCheck(this, ModalPage);

    return _possibleConstructorReturn(this, _getPrototypeOf(ModalPage).apply(this, arguments));
  }

  _createClass(ModalPage, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (prevProps.children !== this.props.children) {
        this.props.updateModalHeight();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          className = _this$props.className,
          header = _this$props.header,
          insets = _this$props.insets,
          platform = _this$props.platform;
      return React.createElement("div", {
        className: classNames(getClassName('ModalPage', platform), className)
      }, React.createElement("div", {
        className: "ModalPage__in-wrap"
      }, React.createElement("div", {
        className: "ModalPage__in"
      }, React.createElement("div", {
        className: "ModalPage__header"
      }, header), React.createElement("div", {
        className: "ModalPage__content"
      }, React.createElement("div", {
        className: "ModalPage__content-in",
        style: isNumeric(insets.bottom) ? {
          paddingBottom: insets.bottom
        } : null
      }, children)))));
    }
  }]);

  return ModalPage;
}(Component);

_defineProperty(ModalPage, "defaultProps", {
  settlingHeight: 75,
  insets: {}
});

export default withInsets(withPlatform(withModalRootContext(ModalPage)));
