// @vkontakte/vkui/dist/components/ModalPage/ModalPage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _getClassName = _interopRequireDefault(require("../../helpers/getClassName"));

var _classNames = _interopRequireDefault(require("../../lib/classNames"));

var _withInsets = _interopRequireDefault(require("../../hoc/withInsets"));

var _utils = require("../../lib/utils");

var _withModalRootContext = _interopRequireDefault(require("../ModalRoot/withModalRootContext"));

var _withPlatform = _interopRequireDefault(require("../../hoc/withPlatform"));

var ModalPage = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(ModalPage, _Component);

  function ModalPage() {
    (0, _classCallCheck2.default)(this, ModalPage);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(ModalPage).apply(this, arguments));
  }

  (0, _createClass2.default)(ModalPage, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (prevProps.children !== this.props.children) {
        this.props.updateModalHeight();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          className = _this$props.className,
          header = _this$props.header,
          insets = _this$props.insets,
          platform = _this$props.platform;
      return _react.default.createElement("div", {
        className: (0, _classNames.default)((0, _getClassName.default)('ModalPage', platform), className)
      }, _react.default.createElement("div", {
        className: "ModalPage__in-wrap"
      }, _react.default.createElement("div", {
        className: "ModalPage__in"
      }, _react.default.createElement("div", {
        className: "ModalPage__header"
      }, header), _react.default.createElement("div", {
        className: "ModalPage__content"
      }, _react.default.createElement("div", {
        className: "ModalPage__content-in",
        style: (0, _utils.isNumeric)(insets.bottom) ? {
          paddingBottom: insets.bottom
        } : null
      }, children)))));
    }
  }]);
  return ModalPage;
}(_react.Component);

(0, _defineProperty2.default)(ModalPage, "defaultProps", {
  settlingHeight: 75,
  insets: {}
});

var _default = (0, _withInsets.default)((0, _withPlatform.default)((0, _withModalRootContext.default)(ModalPage)));

exports.default = _default;
